### ISSUE TITLE
Type of queryArgs in serializeQueryArgs in createApi options is incorrect

### ISSUE THREAD

**Issue by Reedgern** [reduxjs/redux-toolkit#4657]
Hi.
It seems that there is problem with type of queryArgs in serializeQueryArgs of createApi options.
By definition queryArgs is supposed to be the first parameter of baseQuery function.
```
serializeQueryArgs?: SerializeQueryArgs<BaseQueryArg<BaseQuery>>;
```
In reality queryArgs is the parameter received by query function.
I made a [sandbox](https://codesandbox.io/p/sandbox/busy-snyder-p566w9?file=%2Fsrc%2Fapi%2Findex.ts%3A11%2C40) with testApi and added some console.log calls to show what exactly queryArg is.
We can see the supposed type of queryArgs on the first screenshot and we can see the real value on the second one.
![Image](https://github.com/user-attachments/assets/030fefb2-7982-476a-b00b-7dddea29e8d1)
![Image](https://github.com/user-attachments/assets/6f27ab43-80c8-493a-8cbc-9adac8cf9ae8)

**Comment by EskiMojo14** [comment]
yep, that really should just be `SerializeQueryArgs<unknown>` - PR welcome :)

**Comment by markerikson** [comment]
Merged and out in https://github.com/reduxjs/redux-toolkit/releases/tag/v2.3.0 !
